flink-runtime/src/main/java/org/apache/flink/runtime/highavailability/HighAvailabilityServices.java

```diff
@@ -212,19 +213,53 @@ default LeaderRetrievalService getClusterRestEndpointLeaderRetriever() {
     void close() throws Exception;
 
     /**
-     * Closes the high availability services (releasing all resources) and deletes all data stored
-     * by these services in external stores.
+     * Deletes all data stored by high availability services in external stores.
      *
-     * <p>After this method was called, the any job or session that was managed by these high
+     * <p>After this method was called, any job or session that was managed by these high
      * availability services will be unrecoverable.
      *
      * <p>If an exception occurs during cleanup, this method will attempt to continue the cleanup
      * and report exceptions only after all cleanup steps have been attempted.
      *
-     * @throws Exception Thrown, if an exception occurred while closing these services or cleaning
-     *     up data stored by them.
+     * @throws Exception Thrown, if an exception occurred while cleaning up data stored by them.
      */
-    void closeAndCleanupAllData() throws Exception;
+    void cleanupAllData() throws Exception;
+
+    /**
+     * Closes the high availability services (releasing all resources) and optionally deletes all
+     * data stored by these services in external stores.
+     *
+     * <p>After this method was called, any job or session that was managed by these high
+     * availability services will be unrecoverable.
+     *
+     * <p>If an exception occurs during cleanup or close, this method will attempt to continue the
+     * cleanup or close and report exceptions only after all cleanup and close steps have been
+     * attempted.
+     *
+     * @param cleanupData Whether cleanup of data stored by the high availability services should be
+     *     attempted.
+     * @throws Exception Thrown, if an exception occurred while cleaning up data stored by the high
+     *     availability services.
```

Comment:
```suggestion
     * Calls {@link #cleanupAllData()} (if {@code true} is passed as a parameter) before calling {@link #close()} on this instance. Any error that appeared during the cleanup will be propagated after calling {@code close()}.
```
nit: I'm not a big fan of duplicating documentation because it makes maintaining the documentation harder. What do you think of the proposal above?